### ISSUE TITLE
"?submodules=1" Not being acknowledged in Nix Darwin flakes

### ISSUE THREAD

**Issue by PatrickShaw** [LnL7/nix-darwin#847]
The following command does not acknowledge Git submodules: `nix --extra-experimental-features nix-command --extra-experimental-features flakes run nix-darwin -- switch --flake '.?submodules=1'` 

Reproducibility for me is fairly simple. I've created a repository for replicating the behaviour.

1. Go to https://github.com/PatrickShaw/basic_submodule_example
2. Run `nix --extra-experimental-features nix-command --extra-experimental-features flakes run nix-darwin -- switch --flake '.?submodules=1'`
3. You should get an error that includes the following message: `error: getting status of '/nix/store/0ccnxa25whszw7mgbgyzdm4nqc0zwnm8-source/flake.nix': No such file or directory`

This is either a new bug or a regression of #550 

Here's some relevant information about my machine environment:
- MacOS: Sonoma 14.1
- Chip: Apple M3 Pro
- `nix --version`: `nix (Nix) 2.18.1`
- This is a completely new machine. The only thing I've done thus far is install Nix

Thanks reading up on this issue 🙏

**Comment by kivikakk** [comment]
Hi there!

The git repository is not actually in correct shape. What's the contents of your `innermodule` directory? Note that it is not visible on [GitHub](https://github.com/PatrickShaw/basic_submodule_example) (try clicking the link).

Specifically, the contents of `.gitmodules` is incorrect: it's trying to clone the submodule from `https://github.com/PatrickShaw/basic_submodule_example/innermodule`, as the URL in `.gitmodules` is interpreted relative to the clone URL for the repo itself.

In other words, the submodule flake.nix is indeed nowhere to be found.
